anomalib/deploy/inferencers/torch.py

```diff
@@ -23,10 +23,10 @@
 from omegaconf import DictConfig, ListConfig
 from torch import Tensor
 
-from anomalib.core.model import AnomalyModule
-from anomalib.data.transforms.pre_process import PreProcessor
 from anomalib.deploy.optimize import get_model_metadata
 from anomalib.models import get_model
+from anomalib.models.components.anomaly_models import AnomalyModule
```

Comment:
Is `AnomalyModule` a complete module or a base class? I mean can we use it on its own without implementing some methods?

Reply:
If the answer is no, I feel like we could still have it in `anomalib.core`, which would store all the base classes to be implemented.

Reply:
The functions and classes in `anomalib.models.components` are complete implementations and the user could use them off-the-shelf.

Reply:
Agree with @samet-akcay . How about changing anomalib.models.components.anomaly_models.anomaly_models to anomalib.models.base? And may be we can still have dynamic_module under components.

Reply:
Also now that components is under models can we rename it to utils?